sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/LruCache.java

```diff
@@ -0,0 +1,141 @@
+     * If the key-value pair will increase the cache size above the allowed limit the least recently used key-value pair
+     * will be ejected from the cache and the passed key-value pair will be added.
+     *
+     * @param key The key of the value being added to the cache.
+     * @param value The value being added to the cache.
+     * @return The value that was added into the cache.
+     */
+    public synchronized Value put(Key key, Value value) {
+        if (cache.containsKey(key)) {
+            lruKeys.remove(key);
+            lruKeys.add(key);
+            cache.put(key, value);
+        } else {
+            if (cache.size() >= cacheSizeLimit) {
+                cache.remove(lruKeys.poll());
+            }
+
+            cache.put(key, value);
+            lruKeys.add(key);
+        }
+
+        return value;
+    }
+
+    /**
+     * Gets the value associated to the {@code key} from the cache.
+     * <p>
+     * If the cache doesn't contain a value associated to the {@code key} null will be returned.
+     *
+     * @param key The key to get the associated value in the cache.
+     * @return The value associated to the key or null if there is no value associated to the key.
+     */
+    public synchronized Value get(Key key) {
+        if (cache.containsKey(key)) {
+            lruKeys.remove(key);
+            lruKeys.add(key);
+            return cache.get(key);
+        }
+
+        return null;
+    }
+
+    /**
+     * Gets or computes the value for the {@code key}.
+     * <p>
+     * If the {@code key} exists in the cache the value associated with it is return and the key becomes the most
+     * recently used key.
+     * <p>
+     * If the {@code key} doesn't exist, {@code compute} is called with the passed key and the key-value pair is added
+     * into the cache. If the new key-value causes the cache to exceed its allowed size the least recently used
+     * key-value pair is ejected from the cache.
+     *
+     * @param key The key to get the associated value in the cache.
+     * @param compute The function that will supply the value for the key if the key doesn't exist in the cache.
+     * @return Either the found value associated with the key in the cache or the value returned by {@code compute}
+     * when supplied with the {@code key}.
+     */
+    public synchronized Value computeIfAbsent(Key key, Function<Key, Value> compute) {
+        if (cache.containsKey(key)) {
+            lruKeys.remove(key);
+            lruKeys.add(key);
+
+            return cache.get(key);
+        }
+
+        if (cache.size() >= cacheSizeLimit) {
+            cache.remove(lruKeys.poll());
+        }
+
+        Value value = compute.apply(key);
+        cache.put(key, value);
+        lruKeys.add(key);
+
+        return value;
+    }
+
+    /**
+     * Gets the current number of key-value pairs in the cache.
+     *
+     * @return The number of key-value pairs in the cache.
+     */
+    public synchronized int size() {
```

Comment:
We should add `remove(key)` and `clear()` methods to remove a key and clear the whole cache respectively.